### ISSUE TITLE
should we serialize `debug_meta` for transactions?

### ISSUE THREAD

**Issue by marandaneto** [getsentry/sentry-java#1539]
RN transaction comes with:

```
"debug_meta": {
    "transactionSampling": {
      "method": "client_rate",
      "rate": 1
    }
  }
```

during ser/deser, we're losing the debug_meta field

**Comment by marandaneto** [comment]
@rhcarvalho do you know the answer by any chance?

**Comment by jennmueng** [comment]
`debug_meta` is added [here by the JS SDK](https://github.com/getsentry/sentry-javascript/blob/9af1886e7ee6ca7be3f12050b044c73725783575/packages/tracing/src/transaction.ts#L129), and is the type [TransactionMetadata](https://github.com/getsentry/sentry-javascript/blob/9af1886e7ee6ca7be3f12050b044c73725783575/packages/types/src/transaction.ts#L127-L135)

**Comment by bruno-garcia** [comment]
That field is new to me (and doesn't seem to be in the spec): https://develop.sentry.dev/sdk/event-payloads/debugmeta/
Anther one we'd address with #1537 but needs addressing before that to unblock RN.

**Comment by rhcarvalho** [comment]
This is an internal field for used debugging. It is not part of the spec and there is no intention that it will ever be.

For all I can think of, we don't need this field in React Native nor in Java, so the **current behavior of dropping the `debug_meta` field is adequate**.

(@lobsterkatie FYI)

**Comment by lobsterkatie** [comment]
I added this at the behest of @jan-auer. Not sure if he has any plans for it beyond JS.

**Comment by jan-auer** [comment]
We should not be sending this to the server. As @rhcarvalho stated, this was added to the internals of the JavaScript SDK so that the sampling mechanism could be propagated to the `captureEnvelope` function, where it is transformed into an envelope header.

**Comment by lobsterkatie** [comment]
@jan-auer Right. I guess I was more wondering if that `debug_meta` -> envelope header work should happen anywhere besides JS, but it sounds like no.
